File: code/community/1304605478935859208/deepgram-speech-to-text-aws-lambda-websockets.go
package main

import (
	"fmt"
	"os"

	"github.com/deepgram/go-sdk/deepgram"
)

func main() {
	apiKey := os.Getenv("DEEPGRAM_API_KEY")
	filePath := "path/to/your/audiofile.wav"

	client := deepgram.NewClient(apiKey)
	file, err := os.Open(filePath)
	if err != nil {
		fmt.Println("Error opening file:", err)
		return
	}
	defer file.Close()

	options := deepgram.TranscriptionOptions{}
	result, err := client.Transcribe(file, options)
	if err != nil {
		fmt.Println("Error transcribing audio:", err)
		return
	}

	fmt.Println("Transcription:", result)
}

File: code/community/1304605478935859208/deepgram-speech-to-text-aws-lambda-websockets.rs
use deepgram::Deepgram;
use std::env;

#[tokio::main]
async fn main() {
    let api_key = env::var("DEEPGRAM_API_KEY").expect("DEEPGRAM_API_KEY must be set");
    let file_path = "path/to/your/audiofile.wav";

    let dg_client = Deepgram::new(api_key);
    let options = deepgram::TranscriptionOptions::builder()
        .endpoint("https://api.deepgram.com/v1/listen")
        .build();

    match dg_client.transcribe(&options, file_path).await {
        Ok(transcription) => println!("Transcription: {:#?}", transcription),
        Err(e) => eprintln!("Error: {:#?}", e),
    }
}
